Libraries/Share/Share.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule Share
+ * @flow
+ */
+'use strict';
+
+const Platform = require('Platform');
+const {
+  ActionSheetManager,
+  ShareModule
+} = require('NativeModules');
+const invariant = require('fbjs/lib/invariant');
+const processColor = require('processColor');
+
+type Content = { title?: string, message: string } | { title?: string, url: string };
+type Options = { dialogTitle?: string, excludeActivityTypes?: Array<string>, tintColor?: string };
+
+class Share {
+  
+  /**
+   * Open a dialog to share text content.
+   * 
+   * ### Content
+   * 
+   *  - `message` - a message to share
+   *  - `url` - an URL to share. In Android, this will overwrite message
+   *  - `title` - title of the message
+   * 
+   * At least one of URL and message is required.
+   * 
+   * ### Options
+   *
+   * #### iOS
+   *
+   * - `excludedActivityTypes`
+   * - `tintColor`
+   *
+   * #### Android
+   * 
+   * - `dialogTitle`
+   */
+  static share(content: Content, options: ?Options): Promise<Object> {
+    invariant(
+      typeof content === 'object' && content !== null,
+      'Content must a valid object'
+    );
+    invariant(
+      content.url || content.message,
+      'At least one of URL and message is required'
+    );
+    invariant(
+      !content.message || typeof content.message === 'string',
+      'Invalid message: message should be a string.'
+    );
+    invariant(
+      !content.url || typeof content.url === 'string',
+      'Invalid url: url should be a string.'
+    );
+    invariant(
+      !content.title || typeof content.title === 'string',
+      'Invalid title: title should be a string.'
+    );
+
+    if (Platform.OS === 'android') {
+      let dialogTitle = (options !== null && typeof options === 'object' && options.dialogTitle) ? options.dialogTitle : null;
+      return ShareModule.share(content, dialogTitle);
+    } else if (Platform.OS === 'ios') {
```

Comment:
@grabbou this `else if` was a request of @andreicoman11. I guess it's for RN on Windows platform and furthermore.